consensus/votecounter/round_data.go

```diff
@@ -0,0 +1,80 @@
+package votecounter
+
+import (
+	"github.com/NethermindEth/juno/consensus/types"
+	"github.com/NethermindEth/juno/utils"
+)
+
+type roundMap[V types.Hashable[H], H types.Hash, A types.Addr] = map[types.Round]*roundData[V, H, A]
+
+type roundData[V types.Hashable[H], H types.Hash, A types.Addr] struct {
+	proposal               *types.Proposal[V, H, A]
+	uncountedProposerPower types.VotingPower
+	perIDVotes             map[H]*ballotSet[A]
+	nilVotes               ballotSet[A]
+	allVotes               ballotSet[A]
+}
+
+func newRoundData[V types.Hashable[H], H types.Hash, A types.Addr]() roundData[V, H, A] {
+	return roundData[V, H, A]{
+		proposal:               nil,
+		uncountedProposerPower: 0,
+		perIDVotes:             make(map[H]*ballotSet[A]),
+		nilVotes:               newBallotSet[A](),
+		allVotes:               newBallotSet[A](),
+	}
+}
+
+func (r *roundData[V, H, A]) setProposal(proposal *types.Proposal[V, H, A], votingPower types.VotingPower) bool {
+	if r.proposal != nil {
+		return false
+	}
+	r.proposal = proposal
+
+	proposerBallot, ok := r.allVotes.ballots[proposal.Sender]
+	if proposerNotVoted := !ok || (!proposerBallot[Prevote] && !proposerBallot[Precommit]); proposerNotVoted {
+		r.uncountedProposerPower = votingPower
+	}
+	return true
+}
+
+func (r *roundData[V, H, A]) addVote(vote *types.Vote[H, A], votingPower types.VotingPower, voteType VoteType) bool {
```

Comment:
This function only returns `true`, is there a point in it having a explicit return type (perhaps to satisfy an interface?)

Reply:
Good catch! I forgot to use the return result from `perVote.add`.